asteroidsyncservice/watch.cpp

```diff
@@ -133,3 +137,68 @@ void Watch::sendNotify(unsigned int id, QString appName, QString icon, QString b
 {
     m_iface->call("SendNotify", id, appName, icon, body, summary);
 }
+
+bool Watch::screenshotServiceReady()
+{
+    return fetchProperty("StatusScreenshotService").toBool();
+}
+
+unsigned int Watch::screenshotProgress()
+{
+    return m_scrnProgress;
+}
+
+void Watch::screenshotTransferProgress(unsigned int progress)
+{
+    m_scrnProgress = progress;
+    emit screenshotProgressChanged();
+}
+
+void Watch::screenshotReceived(QByteArray data)
+{
+    QString filePath = getScreenshotDir(m_screenshotUrl.dir()) + QDir::separator() + m_screenshotName;
+    QSaveFile file(filePath);
+    file.open(QIODevice::WriteOnly);
+    file.write(data);
+    file.commit();
+    m_screenshotPath = filePath;
+    emit screenshotPathChanged();
+}
+
+QString Watch::screenshotPath() const
+{
+    return m_screenshotPath;
+}
+
+void Watch::setScreenshotUrl(const QString url)
+{
+    QFileInfo fileInfo(url);
+    if(!createDir(fileInfo.dir()))
+        qDebug() << "Unable to create directory";
+    m_screenshotUrl = fileInfo;
+}
+
+bool Watch::createDir(const QDir path)
+{
+    if(!path.exists()) {
+        return path.mkdir(path.path());
+    } else {
+        return true;
+    }
+}
+
+QString Watch::getScreenshotDir(const QDir dir)
+{
+    return dir.path();
+}
+
+QString Watch::createScreenshotFilename(QString filename, const QString suffix)
+{
+    QString fileNameExSuffix = filename.remove(suffix);
+    return "screenshot" + getCurrentDateTime().toString(fileNameExSuffix.remove("screenshot")) + suffix;
```

Comment:
I am not sure I understand why you remove « screenshot » and then add it again.
Also, if m_screenshotUrl is entered by the QML code, why do you make the assumption that it contains « screenshot » ? I can imagine an app that would use qmlasteroid but localize the name of the file so it would be « Capture d’écran du 27/05/2018 » for instance. I don’t understand the logic here. Could you give me more background?
Also, why is the suffix removed and then added again? I’m confused. Can’t QDateTime::toString take an arbitrary string and only replace the format characters?

Reply:
i did not know that i can format the string with inside with ' ' to make clear what needs to replaced and what should stay.
But some more internet searching helped